enhancements/windows-containers/disconnected-support.md

```diff
@@ -0,0 +1,390 @@
+and users [should migrate to IDMSes](https://docs.openshift.com/container-platform/4.14/openshift_images/image-configuration.html#images-configuration-registry-mirror-convert_image-configuration) instead.
+* Using [`ImageContentPolicy` resources](https://docs.openshift.com/container-platform/4.14/rest_api/config_apis/imagecontentpolicy-config-openshift-io-v1.html)
+
+
+## Proposal
+
+### Design Details
+
+WMCO will use the mirrors data in `ImageDigestMirrorSet` and `ImageTagMirrorSet` resources 
+to create a set of [hosts.toml files](https://github.com/containerd/containerd/blob/main/docs/hosts.md#hoststoml-content-description---detail),
+one for each registry mirror, that will be transferred to the Windows instance.
+
+This will be done on during initial node configuration, similarly to how bootstrap files (like the kubelet config file)
+are generated. This way nodes will have the right mirror registry config right when they join the cluster.
+
+
+In addition, the operator should also react to any events on IDMS/ITMS resources through a new "registry controller" --
+when mirror settings are updated, WMCO will then react by:
+- generating an updated set of hosts.toml files
+- publishing them to the windows-services ConfigMap as `Files` 
+
+The windows-services ConfigMap spec will have to be extended:
+* we will add a `content` field the `files` key so it will look like
+```json
+{
+  "path": "filepath",
+  "checksum": "checksum of file, generated at compile time",
+  "content": "base64 encoded file data, similarly to how MCO stores files in MachineConfigs",
+}
+```
+
+WICD, which already watches the services ConfigMap, will react to events on it by copying over the new registry config
+files to each Windows instance, if a file has changed.
+Note that if the newly generated set of files is empty, this implies the cluster mirror regsitry settings have been cleared.
+WMCO will respond by deleting all directories and hosts.toml files under containerd's `config_path` directory.
+
+We can utilize [registry runtime-utils](https://github.com/openshift/runtime-utils/blob/master/pkg/registries/registries.go)
+to easily process and merge the contents of the IDMS/ITMS resources, [as MCO does](https://github.com/openshift/machine-config-operator/blob/master/pkg/controller/container-runtime-config/helpers.go#L407).
+
+
+The directory to save the generated registry config files is defined by the containerd_config.toml's
+`plugins."io.containerd.grpc.v1.cri".registry.config_path`.
+
+The containerd daemon [picks up the registry configuration without requiring restart](https://github.com/containerd/containerd/blob/main/docs/hosts.md#registry-configuration---introduction).
+
+Since WMCO is an day 2 operator (optional, not part of the OCP payload), it will pick up the mirror configuration during
+runtime regardless of whether they were set during cluster install time or at some point during the cluster's lifetime.
```

Comment:
This section feels disjointed, and like an information dump. I suggest adding subheaders to help compartmentalize the info you are giving.

Reply:
On second read, I totally agree. I reworked it and broke it up into separate sections